couchdb/src/main/java/org/hibernate/ogm/datastore/couchdb/dialect/impl/CouchDBTupleSupplier.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.couchdb.dialect.impl;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.hibernate.ogm.dialect.query.spi.ClosableIterator;
+import org.hibernate.ogm.dialect.spi.TransactionContext;
+import org.hibernate.ogm.dialect.spi.TupleSupplier;
+import org.hibernate.ogm.model.spi.Tuple;
+
+
+/**
+ * @author Davide D'Alto
+ */
+public class CouchDBTupleSupplier implements TupleSupplier {
```

Comment:
Maybe we could have a common `TuplesSupplier` taking a list of tuples instead of creating a specific one for CouchDB?